src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/ParsingError.scala

```diff
@@ -0,0 +1,36 @@
+package com.snowplowanalytics.snowplow.analytics.scalasdk
+
+/**
+  * Represents error during parsing TSV event
+  */
+sealed trait ParsingError
+
+object ParsingError {
+
+  /**
+    * Represents error which given line is not a TSV
+    */
+  case object NonTSVPayload extends ParsingError
+
+  /**
+    * Represents error which number of given columns is not equal
+    * to number of expected columns
+    * @param expectedNum expected number of columns
+    * @param gotNum number of given columns
+    */
+  case class ColumnNumberMismatch(expectedNum: Int, gotNum: Int) extends ParsingError
+
+  /**
+    * Represents cases where value in a field is not valid
+    * e.g invalid timestamp, invalid UUID
+    * @param key key of field
+    * @param value value of the field
+    */
+  case class InvalidValue(key: String, value: String) extends ParsingError
```

Comment:
Why not to keep `key` as `Symbol` so far? It seems we're stringifying it everywhere.